### PR TITLE
make dns tests resilient to dns timeouts

### PR DESCRIPTION
**Pull request by akarpz**
### What does this PR do?

creates a new DNS client for each retry of the `require.Eventually`, ensuring a different client ephemeral port is used on each try. the connection found in the tracer will be different and thus have the expected byte counts, timeouts, etc.

I tested this locally by setting the DNS client timeout to 15ms, which is right at the edge of what the request takes on my machine. this makes the request very flakey (often needing to retry 10+ times); but the test still passes even in the worst cases. Also validated that the test does _not_ pass with the same timeout without this change, as the timed out connection is grouped with the successful one and thus the assertions fail.

### Motivation

https://datadoghq.atlassian.net/browse/NPM-3079

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by hmahmood — APPROVED** [review]

File: pkg/network/tracer/tracer_test.go [hunk 1]
```diff
@@ -1013,15 +1013,16 @@ func testDNSStats(t *testing.T, tr *Tracer, domain string, success, failure, tim
...
 	require.Eventually(t, func() bool {
+		dnsClient := new(dns.Client)
+		dnsConn, err := dnsClient.Dial(dnsServerAddr.String())
+		require.NoError(t, err)
+		dnsClientAddr = dnsConn.LocalAddr().(*net.UDPAddr)
 		_, _, err = dnsClient.ExchangeWithConn(queryMsg, dnsConn)
+		_ = dnsConn.Close()
 		return err == nil || timeout != 0
-	}, 3*time.Second, 100*time.Millisecond, "Failed to get dns response")
+	}, 6*time.Second, 100*time.Millisecond, "Failed to get dns response")
```
Comment (leeavital) [inline pkg/network/tracer/tracer_test.go:1025]: why did this change?

**Review by leeavital — COMMENTED** [review]
The assertions here https://github.com/DataDog/datadog-agent/pull/20846/files#diff-e2871eefa3722b7a4b0bb1234dbbb5bb12305f81428382219b1b5d4ba5a67095R1060-R1063 will fail if the client gets a timeout followed by a success

File: pkg/network/tracer/tracer_test.go (hunk 1 above)
Comment (akarpz) [reply pkg/network/tracer/tracer_test.go:1025]: I did this to bring it in line with the same change I made on other DNS retries in tests... basically the client timeout is `2s`, so this gives us 3 potential tries

**Comment by akarpz** [comment]
@leeavital in my description I mention some testing I did locally simulating that case (I think we are talking about the same thing)

**Comment by leeavital** [comment]
Ah my mistake, I read too hastily. Clever workaround!

**Review by leeavital — APPROVED** [review]

**Pull request merged**